scripts/inference/hf_generate.py

```diff
@@ -329,6 +333,15 @@ def _generate(encoded_inp):
             output_tokens = gen_tokens - input_tokens
             total_input_tokens = input_tokens.sum()
             total_output_tokens = output_tokens.sum()
+
+            if total_output_tokens == 0:
+                print('\n\nWarning: No non-special output tokens generated.')
+                print(
+                    'This can happen if the generation only contains padding/eos tokens.'
+                )
+                debug_decoded_gen = tokenizer.batch_decode(
+                    encoded_gen, skip_special_tokens=False)
+                print(f'Debug: Full generation: {debug_decoded_gen}\n\n')
```

Comment:
A couple notes on this:
- As written, when this triggers it will re-print all the prompts which were already printed above.
- This won't trigger in the case where the batch has at least 1 prompt that led to a non-empty response.

So, I recommend folding this logic into the the block where it prints the prompt+continuation for each sequence in the batch. Ideally, it would see that `continuation` is empty and then add the printouts. I don't think we need to print the full generation with all the special tokens -- that could easily end up being just a flood of EOS in the batched setting.

Reply:
Definitely a more elegant solution. I just wanted to throw this in with the tokenization fix since it currently just throws the `divide_by_zero` warning. I'll include this but I also will open a separate PR with more changes to `hf_generate` soon to handle padding, delimiters etc.